src/cpp/database/database.hpp

```diff
@@ -150,6 +150,24 @@ class Database
             EntityKind entity_type,
             const EntityId& entity_id) const;
 
+    /**
+     * Get all entities of a given EntityKind containing the provided attribute
+     *
+     * If the given EntityKind does not contain an attribute of the provided type,
+     * BadParameter is thrown.
+     *
+     * @param entity_kind The EntityKind of the fetched entities
+     * @param attribute The attribute of the entities to search for
+     * @return A vector containing the EntityIds of the entities
+     */
+    template <typename T>
+    std::vector<EntityId> get_entities_with_attribute(
+            EntityKind entity_kind,
+            const T& attribute) const
```

Comment:
How is this going to make a difference between GUID and name? Both are plain `std::string`